src/main/java/com/atlassian/jira/cloud/jenkins/config/ConfigManagementLink.java

```diff
@@ -0,0 +1,319 @@
+            return "";
+        }
+        return formData.getJSONObject(key).optString(key + "Regex", "");
+    }
+
+    @VisibleForTesting
+    void SendConfigDataToJira(final JSONObject formData) {
+        JSONArray sitesArray = formData.getJSONArray("sites");
+
+        String ipAddress = getIpAddress();
+
+        boolean autoBuildsEnabled = formData.has("autoBuilds");
+        boolean autoDeploymentsEnabled = formData.has("autoDeployments");
+        String autoBuildsRegex = getRegexFromFormData(formData, "autoBuilds");
+        String autoDeploymentsRegex = getRegexFromFormData(formData, "autoDeployments");
+
+        for (Object siteObject : sitesArray) {
+            JSONObject site = (JSONObject) siteObject;
+            String webhookUrl = site.getString("webhookUrl");
+            String credentialsId = site.getString("credentialsId");
+
+            final Optional<String> maybeSecret = secretRetriever.getSecretFor(credentialsId);
+
+            try {
+                this.pluginConfigApi.sendConnectionData(
+                        webhookUrl,
+                        maybeSecret.get(),
+                        ipAddress,
+                        autoBuildsEnabled,
+                        autoBuildsRegex,
+                        autoDeploymentsEnabled,
+                        autoDeploymentsRegex,
+                        PipelineLogger.noopInstance());
+
+            } catch (Exception e) {
+                LOGGER.warning("Failed to send Data to Jenkins for Jira");
+            }
+        }
+    }
+
+    // Incomplete sites or Deleted sites are marked with active=false on the client side, we want to
+    // remove them from the JSON object
+    @VisibleForTesting
+    JSONObject removeInvalidSites(final JSONObject formData) {
+        if (formData.has("sites")) {
+            Object sites = formData.get("sites");
+            if (sites instanceof JSONArray) {
+                JSONArray sitesArray = (JSONArray) sites;
+
+                for (int i = 0; i < sitesArray.size(); i++) {
+                    JSONObject siteObject = sitesArray.getJSONObject(i);
+                    if (siteObject.has("active")
+                            && siteObject.optString("active").equals("false")) {
+                        sitesArray.remove(i);
+                        i--;
+                    }
+                }
+                formData.put("sites", sitesArray);
+            }
+        }
+        return formData;
+    }
+
+    @RequirePOST
+    public void doSaveConfiguration(final StaplerRequest req, final StaplerResponse res)
+            throws ServletException, IOException, Descriptor.FormException {
+        JSONObject formData = req.getSubmittedForm();
+        JSONObject transformedFormData = removeInvalidSites(formData);
+
+        // Failure to send config data should not stop config save, we also dont need to wait for it to complete
+        CompletableFuture.runAsync(() -> SendConfigDataToJira(transformedFormData));
```

Comment:
Fair, ive wrapped the config actiona in a try catch, but the SendConfigDataToJira is wrapped up in that async because 1, its slow! but also, if it fails I dont want it to stop the rest of the actions happening